swap_meet/vendor.py

```diff
@@ -0,0 +1,100 @@
+
+class Vendor:
+    def __init__(self, inventory = None): 
+        if inventory == None:
+            self.inventory = []      # instances
+        else:
+            self.inventory = inventory
+            
+    # call other instances with self.
+    def add(self, item):   # adds item to inventory and returns item
+        self.inventory.append(item)
+        return item
+            
+    def remove(self, item): # removes item from inventory and returns item
+        if item not in (self.inventory):
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+    
+    def get_by_category(self, category): # puts items in inventory list by category
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+
+
+
+    def swap_items(self, friend, my_item, their_item): # swap my item and friend item
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False 
+
+        #refactored 
+        self.remove(my_item)
+        friend.add(my_item)
+        
+        friend.remove(their_item)
+        self.add(their_item)
+        return True
+
+
+        # # orginal version below
+        # self.inventory.remove(my_item)
+        # friend.inventory.append(my_item)
+        # friend.inventory.remove(their_item)
+        # self.inventory.append(their_item)
+        # return True 
+
+
+
+    def swap_first_item(self, friend):   # swap our first items
+        # refactored
+        if len(self.inventory) <= 0 or len(friend.inventory) <= 0:  # if inventory is empty return False
+            return False 
+        
+        # refactored code 
+        my_first_item = self.inventory[0]  # first item in inventory
+        friend_first_item = friend.inventory[0] # friends first item in inventory
+
+        self.swap_items(friend, my_first_item, friend_first_item)
+        return True
```

Comment:
👍 this is great! I can make on suggestion! we can save some variables:

```suggestion
    def swap_first_item(self, friend):   # swap our first items
        # refactored
        if len(self.inventory) <= 0 or len(friend.inventory) <= 0:  # if inventory is empty return False
            return False 
        # refactored code 
        self.swap_items(friend, self.inventory[0], friend.inventory[0])
        return True
```